### PR TITLE
Added frontend connector

### PR DESCRIPTION
**Pull request by matt712**

### CODE REVIEW

**Comment by matt712** [comment]
Journey tests are failing

File: test/controllers/dividends/DividendsCYAControllerSpec.scala [hunk 1]
```diff
@@ -145,4 +150,48 @@ class DividendsCYAControllerSpec extends ViewTest {
...
+      Some(10),
+      otherUkDividends = Some(true),
+      Some(10)
+    )
+    "redirect to the overview page" when {
+
+      "there is session data " in new TestWithAuth {
+        val request: FakeRequest[AnyContentAsEmpty.type] = FakeRequest().withSession(
+          SessionValues.CLIENT_NINO -> "someNino".toString(),
+          SessionValues.CLIENT_MTDITID -> Json.toJson("someMtdItd").toString(),
```
Comment (robert-webb-hmrc) [inline test/controllers/dividends/DividendsCYAControllerSpec.scala:166]: typo here

File: test/controllers/dividends/DividendsCYAControllerSpec.scala [hunk 2]
```diff
@@ -145,4 +150,48 @@ class DividendsCYAControllerSpec extends ViewTest {
...
+          .returning(Future.successful(Right(DividendsResponseModel(204))))
+
+        val result: Future[Result] = controller.submit(2020)(request)
+        status(result) shouldBe SEE_OTHER
+        redirectUrl(result) shouldBe mockAppConfig.incomeTaxSubmissionOverviewUrl(2020)
+      }
+      "service returns left" in new TestWithAuth {
+        val request: FakeRequest[AnyContentAsEmpty.type] = FakeRequest().withSession(
+          SessionValues.CLIENT_NINO -> "someNino",
+          SessionValues.CLIENT_MTDITID -> Json.toJson("someMtdItd").toString(),
```
Comment (robert-webb-hmrc) [inline test/controllers/dividends/DividendsCYAControllerSpec.scala:181]: as above

File: test/services/DividendsSubmissionServiceSpec.scala [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        Some(10),
+        otherUkDividends = Some(true),
+        Some(10)
+      )
+      val dsmData = DividendsSubmissionModel(
+        Some(10),
+        Some(10)
+      )
+      val nino = "someNino"
+      val mtdItid = "SomeMtdItd"
```
Comment (robert-webb-hmrc) [inline test/services/DividendsSubmissionServiceSpec.scala:49]: typo here

File: app/services/DividendsSubmissionService.scala [hunk 4]
```diff
@@ -0,0 +1,36 @@
+/*
```
Comment (robert-webb-hmrc) [inline app/services/DividendsSubmissionService.scala:1]: Could we review the coverage on the submission service? 
Screenshot 2020-11-23 at 16 29 42
Screenshot 2020-11-23 at 16 29 50

File: app/controllers/dividends/DividendsCYAController.scala [hunk 5]
```diff
@@ -76,10 +82,16 @@ class DividendsCYAController @Inject()(
...
   }
 
-  def submit(taxYear: Int): Action[AnyContent] = authorisedAction { implicit user =>
-    //TODO To be used for the submission
+  def submit(taxYear: Int): Action[AnyContent] = authorisedAction.async { implicit user =>
     val cyaData: Option[DividendsCheckYourAnswersModel] = getSessionData[DividendsCheckYourAnswersModel](SessionValues.DIVIDENDS_CYA)
-    Redirect(appConfig.incomeTaxSubmissionOverviewUrl(taxYear))
+   user.session.get(SessionValues.CLIENT_NINO) match {
+      case Some(nino) => dividendsSubmissionService.submitDividends(cyaData, nino, user.mtditid, taxYear).map(response => response match {
+        case Right(DividendsResponseModel(NO_CONTENT)) => Redirect(appConfig.incomeTaxSubmissionOverviewUrl(taxYear)) //todo error handling
```
Comment (robert-webb-hmrc) [inline app/controllers/dividends/DividendsCYAController.scala:89]: is this todo part of this ticket? would be better to remove it if not.

**Review by robert-webb-hmrc — CHANGES_REQUESTED** [review]

File: test/controllers/dividends/DividendsCYAControllerSpec.scala [hunk 6]
```diff
@@ -145,4 +150,48 @@ class DividendsCYAControllerSpec extends ViewTest {
...
```
Comment (robert-webb-hmrc) [inline test/controllers/dividends/DividendsCYAControllerSpec.scala:181]: One typo left

**Review by robert-webb-hmrc — CHANGES_REQUESTED** [review]

**Review by robert-webb-hmrc — APPROVED** [review]

**Pull request merged**